### ISSUE TITLE
python: docker: build failure on Darwin

### ISSUE THREAD

**Issue by smaret** [NixOS/nixpkgs#75355]
**Describe the bug**

`python37Packages.docker` fails to build on Darwin with the following error message:

```python
builder for '/nix/store/320aynm8i2bnks3ijcyhv69x6k9bcvbf-python3.7-docker-4.1.0.drv' failed with exit code 1; last 10 log lines:
          self.addCleanup(shutil.rmtree, base)
          for d in ['foo', 'bar']:
              os.makedirs(os.path.join(base, d))
          sock = socket.socket(socket.AF_UNIX)
          self.addCleanup(sock.close)
  >       sock.bind(os.path.join(base, 'test.sock'))
  E       OSError: AF_UNIX path too long
```

**To Reproduce**

1. `nix build --option build-use-sandbox true -I . nixpkgs.python37Packages.docker`

**Metadata**

 - system: `"x86_64-darwin"`
 - host os: `Darwin 19.0.0, macOS 10.15.1`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.1`
 - channels(smaret): `"nixpkgs-19.03pre170546.0f43790d2ce"`
 - nixpkgs: `/nix/store/kd1bflipda7jjnfqxjfiym9268akzq2c-master.tar.gz`

cc @jonringer

**Comment by jonringer** [comment]
unfortunately the sandbox environment does create really long paths, too long for darwin to handle by default, I'll disable the affected tests

**Comment by marsam** [comment]
you could backport https://github.com/NixOS/nixpkgs/pull/72002/; however I recommend you to upgrade to 19.09, because 19.03 has reached end of support:
>End of support is planned for end of October 2019, handing over to 19.09.
https://nixos.org/nixos/manual/release-notes.html#sec-release-19.03-highlights

**Comment by smaret** [comment]
@marsam I'm using the HEAD version of `nixpkgs` from Github, not the 19.03  channel (not sure why `nix-info` says that I'm using 19.03). It's a different issue than the one you fixed with #72002.
